test/pollingInterval.js

```diff
@@ -1244,3 +1245,385 @@ test('Polling schemas (subscriptions should be handled)', async t => {
+        __resolveReference: (user) => user
+      }
+    }
+  })
+
+  await userService.listen({ port: 0 })
+
+  const userServicePort = userService.server.address().port
+
+  const posts = {
+    p1: {
+      pid: 'p1',
+      title: 'Post 1',
+      content: 'Content 1',
+      authorId: 'u1'
+    }
+  }
+
+  postService.register(GQL, {
+    schema: buildFederationSchema(`
+      type Post @key(fields: "pid") {
+        pid: ID!
+        author: User
+      }
+
+      extend type Query {
+        topPosts(count: Int): [Post]
+      }
+
+      type User @key(fields: "id") @extends {
+        id: ID! @external
+        topPosts(count: Int!): [Post]
+      }`),
+    resolvers: {
+      Post: {
+        __resolveReference: (post, args, context, info) => {
+          return posts[post.pid]
+        },
+        author: (post, args, context, info) => {
+          return {
+            __typename: 'User',
+            id: post.authorId
+          }
+        }
+      },
+      User: {
+        topPosts: (user, { count }, context, info) => {
+          return Object.values(posts)
+            .filter((p) => p.authorId === user.id)
+            .slice(0, count)
+        }
+      },
+      Query: {
+        topPosts: (root, { count = 2 }) => Object.values(posts).slice(0, count)
+      }
+    }
+  })
+
+  await postService.listen({ port: 0 })
+
+  const postServicePort = postService.server.address().port
+
+  const services = [
+    {
+      name: 'user',
+      url: `http://localhost:${userServicePort}/graphql`
+    }
+  ]
+
+  const servicesFn = async () => services
+  await gateway.register(plugin, {
+    gateway: {
+      services: servicesFn,
+      pollingInterval: 2000
+    }
+  })
+
+  const res = await gateway.inject({
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    url: '/graphql',
+    body: JSON.stringify({
+      query: `
+        query MainQuery {
+          me {
+            id
+            name
+          }
+        }
+      `
+    })
+  })
+
+  t.same(JSON.parse(res.body), {
+    data: {
+      me: {
+        id: 'u1',
+        name: 'John'
+      }
+    }
+  })
+
+  const res2 = await gateway.inject({
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    url: '/graphql',
+    body: JSON.stringify({
+      query: `
+        query MainQuery {
+          topPosts {
+            pid
+          }
+        }
+      `
+    })
+  })
+
+  t.same(JSON.parse(res2.body), {
+    errors: [
+      {
+        message: 'Cannot query field "topPosts" on type "Query".',
+        locations: [{ line: 3, column: 11 }]
+      }
+    ],
+    data: null
+  })
+
+  services.push({
+    name: 'post',
+    url: `http://localhost:${postServicePort}/graphql`
+  })
+
+  for (let i = 0; i < 10; i++) {
+    await t.context.clock.tickAsync(200)
+  }
+
+  // We need the event loop to actually spin twice to
+  // be able to propagate the change
+  await immediate()
+  await immediate()
```

Comment:
given the added async operations, you might need more of these.

Reply:
Would it be okay to conditionally loop in the test until it either checks or times out?

Reply:
let's try that!